src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3499,6 +3557,52 @@ namespace winrt::TerminalApp::implementation
         return profile;
     }
 
+    // Function Description:
+    // - Helper to launch a new WT instance elevated. It'll do this by asking
+    //   the shell to elevate the process for us. This might cause a UAC prompt.
+    //   The elevation is performed on a background thread, as to not block the
+    //   UI thread.
+    // Arguments:
+    // - newTerminalArgs: A NewTerminalArgs describing the terminal instance
+    //   that should be spawned. The Profile should be filled in with the GUID
+    //   of the profile we want to launch.
+    // Return Value:
+    // - <none>
+    // Important: Don't take the param by reference, since we'll be doing work
+    // on another thread.
+    fire_and_forget TerminalPage::_OpenElevatedWT(NewTerminalArgs newTerminalArgs)
+    {
+        // Hop to the BG thread
+        co_await winrt::resume_background();
+
+        // This will get us the correct exe for dev/preview/release. If you
+        // don't stick this in a local, it'll get mangled by ShellExecute. I
+        // have no idea why.
+        const auto exePath{ GetWtExePath() };
+
+        // Build the commandline to pass to wt for this set of NewTerminalArgs
+        winrt::hstring cmdline{
+            fmt::format(L"new-tab {}", newTerminalArgs.ToCommandline().c_str())
+        };
+
+        // Build the args to ShellExecuteEx. We need to use ShellExecuteEx so we
+        // can pass the SEE_MASK_NOASYNC flag. That flag allows us to safely
+        // call this on the background thread, and have ShellExecute _not_ call
+        // back to us on the main thread. Without this, if you close the
+        // Terminal quickly after the UAC prompt, the elevated WT will never
+        // actually spawn.
```

Comment:
Wait I thought that's what the `Sleep(5000)` was for... are there multiple timeout problems here?